src/controller/CHIPDeviceController.cpp

```diff
@@ -938,6 +939,21 @@ CHIP_ERROR DeviceCommissioner::StopPairing(NodeId remoteDeviceId)
     return CHIP_NO_ERROR;
 }
 
+void DeviceCommissioner::CancelCommissioningInteractions()
+{
+    if (mReadClient)
+    {
+        ChipLogDetail(Controller, "Cancelling read request for step '%s'", StageToString(mCommissioningStage));
+        mReadClient = nullptr; // destructor cancels
```

Comment:
Yes, I can change this to `reset()`. Given that smart pointers aren't super common in the code base I agree it's more readable to use methods rather than assignments.